### PR TITLE
fix: no infinite running command notification for diff file and folder commands

### PR DESCRIPTION
**Pull request by CristiCanizales**
### What does this PR do?
Gets rid of the Running SFDX: Diff File Against Org notification when the diff source fails.

### What issues does this PR fix or reference?
@W-16006725@

### Functionality Before
- Running notification even when the command already failed
image

### Functionality After
- Only failure notifications when it fails
image

### CODE REVIEW

File: packages/salesforcedx-vscode-core/src/commands/sourceDiff.ts [hunk 1]
```diff
@@ -109,7 +109,7 @@ export const handleCacheResults = async (
     }
   } else {
     const message = nls.localize('source_diff_components_not_in_org');
-    await notificationService.showErrorMessage(message);
+    notificationService.showErrorMessage(message);
```
Comment (peternhale) [inline packages/salesforcedx-vscode-core/src/commands/sourceDiff.ts:112]: Please use `void notificationService.showErrorMessage(message);`

Comment (peternhale) [reply packages/salesforcedx-vscode-core/src/commands/sourceDiff.ts:112]: Also, on line 108, the await is not necessary, given `differ.diffFolder` does not return a promise.

**Review by peternhale — CHANGES_REQUESTED** [review]

File: packages/salesforcedx-vscode-core/src/commands/sourceDiff.ts (hunk 1 above)
Comment (CristiCanizales) [reply packages/salesforcedx-vscode-core/src/commands/sourceDiff.ts]: good catch! done

**Review by daphne-sfdc — APPROVED** [review]
Approved! 🎉 

1. Diff file not present in the org ✅ 
Screenshot 2024-08-14 at 8 08 17 AM

2. Diff folder with no files present in the org ✅ 
Screenshot 2024-08-14 at 1 20 53 PM

3. Diff folder with some files present in the org -> only the files that are present in the org are listed in the diff ✅

**Review by peternhale — APPROVED** [review]

**Pull request merged**